openapi.yaml

```diff
@@ -1023,6 +1055,74 @@ components:
                   description: Job reason code
                   example: ABCDE
                   nullable: true
+    DeductionsResource:
+      type: object
+      properties:
+        id:
+          type: string
+          description: OSU ID of a employee
+          pattern: ^\d{9}$
+        type:
+          $ref: '#/components/schemas/DeductionsType'
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        attributes:
+          properties:
+            responseMsg:
+              type: string
+              description: A payroll deductions success or failure message.
+              example: SUCCESS
+    DeductionsResult:
+      type: object
+      properties:
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        data:
+          $ref: '#/components/schemas/DeductionsResource'
+    DeductionsPostBody:
+      type: object
+      required:
+        - data
+      properties:
+        data:
+          type: object
+          required:
+            - type
+          additionalProperties: false
+          properties:
+            type:
+              $ref: '#/components/schemas/DeductionsType'
+            attributes:
+              type: object
+              additionalProperties: false
+              required:
+                - osuId
+                - effectiveDate
+                - position
+                - suffix
+              properties:
+                osuId:
+                  type: string
+                  description: OSU ID of an employee
+                  example: '123456789'
+                  pattern: ^\d{9}$
+                  nullable: false
+                effectiveDate:
+                  type: string
+                  format: date
+                  description: ServiceNow job effective date. Format is 'YYYY-MM-DD'.
+                  example: '1970-07-08'
+                  nullable: false
+                positionNumber:
+                  type: string
+                  description: The job position number
+                  example: 'C12345'
+                  nullable: false
+                suffix:
```

Comment:
Are deductions related to jobs? I'm a little confused now. Would a deduction resource bind with an employee, a job or both? This could result in a very different endpoint design.

Reply:
Deductions are for the employee, but they have conditions that are also based on the job. For example, one of the deduction codes depends on the employee's current location, which is pulled from the job location. The actual deduction isn't tied technically tied to the job though.

Reply:
It's also important to make sure that the employee has an active position before applying payroll deductions. Otherwise someone could be enrolled in benefits and become eligible for insurance without actually having a job at OSU.

Reply:
So, to clarify:

1. Deductions must be tied to an employee.
2. Deductions are not tied to the job, but the job info is "required" to be served as a condition.
3. An employee can have more than one deduction for one job.

Are they correct? Also, a follow-up question: is the only "active job" to be used as the condition?

Reply:
1. yes
2. yes
3. yes, we add 13 different deductions to an employee when they start a new job

follow up questions:
The job on the onboarding case needs to be used as the condition. Employees can have multiple active jobs, and can be in the onboarding process for multiple jobs at the same time. This is why I'm passing in the position/suffix from the onboarding case, in order to ensure we're adding deductions based on the correct position. In a perfect world the job location for each position would be the same for all active jobs, but data entry is not perfect and employee reporting is not perfect, so I need to use the onboarding case job location as the source of truth. They also could have a current active job with deductions based on one location, but be transitioning to a new job with the new effective date and have that job location change. Which would show 2 active jobs with different locations in Banner.